props/saber_BC_buttons.h

```diff
@@ -386,31 +1134,317 @@ EFFECT(volmax);     // for maximum volume reached
+  }
+
+  void update() override {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    // Overridden to substitute the tick sound
+    this->say_time_ = Cyclint<uint32_t>(millis()) + 1;
+    if (!this->say_time_) this->say_time_ += 1;
+    this->fadeout(SaberBase::sound_length);
+  }
+
+  void select() override {
+    cancelling_ = false;
+    mode::getSL<SPEC>()->SaySave();
+    SPEC::SteppedMode::select();
+  }
+
+  void exit() override {
+    PVLOG_NORMAL << "** Exit Volume Menu\n";
+    cancelling_ = true;
+    SPEC::MenuBase::exit();
+  }
+
+  bool mode_Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, 0)) {
+      // Custom button controls for BCVolumeMode
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMaxVolume();
+        return true;
+
+      case EVENTID(BUTTON_POWER, EVENT_SECOND_HELD_MEDIUM, 0):
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMinVolume();
+        return true;
     }
+    // Use the select and exit controls from SelectCancelMode
+    return SPEC::SelectCancelMode::mode_Event2(button, event, modifiers);
+  }
+  bool cancelling_ = false;
+};
+
+#ifdef DYNAMIC_BLADE_LENGTH
+
+int selected_blade_;
+
+#ifdef BC_DUAL_BLADES
+template<class SPEC>
+struct BCSelectBladeMode : public SPEC::MenuBase {
+  int current_blade_ = 1; // Start with the first blade
+  ShowColorSingleBladeTemplate<Pulsing<ShowColorStyle,Black,800>,Pulsing<ShowColorStyle,Black,800>> highlighted_blade_;
+
+  uint16_t size() override {
+    return NUM_BLADES;
+  }
+  void mode_activate(bool onreturn) override {
+    mode::getSL<SPEC>()->SayEditBladeLength();
+    mode::getSL<SPEC>()->SaySelectBlade();
+    ShowColorStyle::SetColor(Color16(65535, 65535, 65535));
+    SPEC::SteppedMode::mode_activate(onreturn);
+    highlighted_blade_.Start(current_blade_);
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade_ << "\n";
+    say();
+  }
+
+  void next() override {
+    highlighted_blade_.Stop(current_blade_);
+    current_blade_ = (current_blade_ % NUM_BLADES) + 1;
+    highlighted_blade_.Start(current_blade_);
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade_ << "\n";
+    say();
+  }
+
+  void prev() override {
+    highlighted_blade_.Stop(current_blade_);
+    current_blade_ = (current_blade_ + NUM_BLADES - 2) % NUM_BLADES + 1;
+    highlighted_blade_.Start(current_blade_);
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade_ << "\n";
+    say();
+  }
+
+  void say() override {
+    mode::getSL<SPEC>()->SayBlade();
+    mode::getSL<SPEC>()->SayWhole(current_blade_);
+  }
+
+  void select() override {
+    // Set the current blade for the ChangeBladeLengthMode before transitioning
+    // Deactivate current mode and push to LED length adjustment mode for the selected blade
+    selected_blade_ = current_blade_;
+    highlighted_blade_.Stop(current_blade_);
+    SPEC::SteppedMode::mode_deactivate();
```

Comment:
There is no need to call mode_deactivate explicitly, pop_mode takes care of that.

Reply:
cool. removed.